hw08-json-writer/src/main/java/ru/otus/json/writer/gson/GsonWriter.java

```diff
@@ -0,0 +1,79 @@
+package ru.otus.json.writer.gson;
+
+import java.util.*;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+public class GsonWriter {
+
+    //Строим строку из массива.
+    private StringBuilder arrayToGson(Object object) throws IllegalAccessException {
+        var sb = new StringBuilder();
+        for (int i = 0; i < Array.getLength(object); i++ ) {
+            Object obj = Array.get(object, i);
+            sb.append(gsonBuilder(obj));
+        }
+        return GsonStringBuild.arrayWraper(sb);
+    }
+
+    //Строим строку из коллекции.
+    private StringBuilder collectionToGson(Object object) throws IllegalAccessException {
+        Object[] collectionObj = ((Collection) object).toArray();
+        return arrayToGson(collectionObj);
+    }
+
+    //Строим JSON строку.
+    private StringBuilder gsonBuilder(Object object) throws IllegalAccessException {

```

Comment:
Если приходится комментировать метод, то лучше назвать метод по другому, чтобы он не нуждался в комментарии.
```suggestion
    private StringBuilder buildJsonString(Object object) throws IllegalAccessException {
```

А, если пишете комментарии к методам, то лучше их сразу писать как javadoc

Reply:
Поправил.